src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -111,7 +121,15 @@ void RGWOp_User_Info::execute(optional_yield y)
   op_state.set_fetch_stats(fetch_stats);
   op_state.set_sync_stats(sync_stats);
 
-  op_ret = RGWUserAdminOp_User::info(s, driver, op_state, flusher, y);
+  // dump_keys is false if users-without-keys is 'read' and
+  // the user is not the system user
+  int keys_perm = s->user->get_info().caps.check_cap("users-without-keys", RGW_CAP_READ);
+  if (keys_perm == 0 && !op_state.system) {
```

Comment:
in addition to system users, admin users should also see the keys by default. you can test for admin access with something like `s->auth.identity->is_admin_of(uid)`

Reply:
What is an admin user? A user with the cap `users` set that created a user? I can't find any upstream documentation on that.